.changeset/rich-lemons-think.md

```diff
@@ -0,0 +1,5 @@
+---
+"@hashicorp/design-system-components": patch
+---
+
+`CodeBlock` - fixed line numbers always shown when `highlightLines` is set
```

Comment:
Perhaps something more like:

"`CodeBlock` -  uncoupled display of line numbers when `highlightLines` is set"